src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -1665,5 +1666,73 @@ CHIP_ERROR ExtractPubkeyFromX509Cert(const ByteSpan & certificate, Crypto::P256P
     return err;
 }
 
+CHIP_ERROR ExtractAKIDFromX509Cert(const ByteSpan & certificate, MutableByteSpan & akid)
+{
+    CHIP_ERROR err                       = CHIP_NO_ERROR;
+    X509 * x509certificate               = nullptr;
+    const unsigned char * pCertificate   = certificate.data();
+    const unsigned char ** ppCertificate = &pCertificate;
+    const ASN1_OCTET_STRING * akidString = nullptr;
+
+    x509certificate = d2i_X509(NULL, ppCertificate, static_cast<long>(certificate.size()));
+    VerifyOrExit(x509certificate != nullptr, err = CHIP_ERROR_NO_MEMORY);
+
+    akidString = X509_get0_authority_key_id(x509certificate);
+
+    VerifyOrExit(akidString->length == static_cast<int>(akid.size()), err = CHIP_ERROR_INVALID_MESSAGE_LENGTH);
+
+    memcpy(akid.data(), akidString->data, static_cast<size_t>(akidString->length));
+
+exit:
+    X509_free(x509certificate);
+
+    return err;
+}
+
+CHIP_ERROR ExtractVIDFromX509Cert(const ByteSpan & certificate, VendorId & vid)
+{
+    CHIP_ERROR err                       = CHIP_NO_ERROR;
+    X509 * x509certificate               = nullptr;
+    const unsigned char * pCertificate   = certificate.data();
+    const unsigned char ** ppCertificate = &pCertificate;
+    constexpr char vidNeedle[]           = "1.3.6.1.4.1.37244.2.1";
```

Comment:
Suggest expressing where you got this, and what its symbolic name is.

Reply:
will update